locations/spiders/nissan_za.py

```diff
@@ -0,0 +1,46 @@
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class NissanZASpider(scrapy.Spider):
+    name = "nissan_za"
+    item_attributes = {"brand": "Nissan", "brand_wikidata": "Q20165"}
+
+    def start_requests(self):
+        yield scrapy.Request(
+            url="https://eu.nissan-api.net/v2/publicAccessToken?locale=en_ZA&scope=READ&proxy=*&brand=nissan&environment=prod",
+            headers={"origin": "https://www.nissan.co.za"},
```

Comment:
This should iterate across all the combinations of locales and origins for "Find My Dealer" pages linked from each country webpage at https://www.nissan-europe.com/

They all appear to be using the same eu.nissan-api.net API and thus this spider can be readily expanded beyond 3 countries to 20 or whatever it ends up being.

Reply:
It turns out that the origin is not too important, only the locale seems to determine what dealers are returned.

I'm now looping over a list of locales I found in the `locale` package. A lot of them get 400 responses and no dealers so the requests feel wasted. Is this a good approach or would it be better to maintain a list of known working locales in the source that we update over time?